src/adams_methods.jl

```diff
@@ -0,0 +1,655 @@
+################################################################################
+# Fixed Step Adams-Bashforth multi-step solver for nonstiff problems
+#
+#     [T,X] = ode_ms(ODEFUN, Y0, TSPAN,ORDER=4) with TSPAN = [T0:H:TFINAL]
+#
+#     Note: If method is run at order k, the first step is taken at order 1,
+#     and the order increases each step until it is order k on the kth step.
+#     Thus results may be of lower accuracy than expected.
+#     (Main ref: Hairer & Wanner 1996, Vol I, p.359-360, 372)
+################################################################################
+
+function ode_ms(F, y0, tspan, order::Integer)
+    h = diff(tspan)
+    y = Array(typeof(y0), length(tspan))
+    y[1] = y0
+
+    if 1 <= order <= 4
+        b = ms_coefficients4
+    else
+        b = zeros(order, order)
+        b[1:4, 1:4] = ms_coefficients4
+        for s = 5:order
+            for j = 0:(s - 1)
+                # Assign in correct order for multiplication below
+                #  (a factor depending on j and s) .* (an integral of a polynomial with -(0:s), except -j, as roots)
+                p_int = polyint(poly(diagm(-[0:j - 1; j + 1:s - 1])))
+                b[s, s - j] = ((-1)^j / factorial(j)
+                               / factorial(s - 1 - j) * polyval(p_int, 1))
+            end
+        end
+    end
+
+    # TODO: use a better data structure here (should be an order-element circ buffer)
+    dy = similar(y)
+    for i = 1:length(tspan)-1
+        # Need to run the first several steps at reduced order
+        steporder = min(i, order)
+        dy[i] = F(tspan[i], y[i])
+
+        y[i+1] = y[i]
+        for j = 1:steporder
+            y[i+1] += h[i]*b[steporder, j]*dy[i-(steporder-1) + (j-1)]
+        end
+    end
+    return vcat(tspan), y
+end
+
+# Use order 4 by default
+ode4ms(F, y0, tspan) = ODE.ode_ms(F, y0, tspan, 4)
+ode5ms(F, y0, tspan) = ODE.ode_ms(F, y0, tspan, 5)
+
+
+################################################################################
+#  Fixed Step Adams-Moulton PECE solver for nonstiff problems
+#
+#     [T,X] = ODE_AM(ODEFUN, Y0, TSPAN,ORDER=4) with TSPAN = [T0:H:TFINAL]
+#
+#     Note: If method is run at order k, the first step is taken at order 1,
+#     and the order increases each step until it is order k on the kth step.
+#     Thus results may be of lower accuracy than expected.
+#     (Main ref: Hairer & Wanner 1996, Vol I, p.359-360, 372)
```

Comment:
Maybe say: "A possible solution to this is to take very small first steps 1:order."